torch/lib/c10d/reducer.cpp

```diff
@@ -1466,18 +1475,61 @@ void Reducer::ensure_prior_reduction_finished() {
   }
 }
 
+void Reducer::set_env_variables() {
+  // Environment variables
+  ddp_logging_data_->master_port = parse_env("MASTER_PORT");
+  ddp_logging_data_->master_addr = parse_env("MASTER_ADDR");
+  ddp_logging_data_->cuda_visible_devices = parse_env("CUDA_VISIBLE_DEVICES");
+  ddp_logging_data_->gloo_socket_ifname = parse_env("GLOO_SOCKET_IFNAME");
+  ddp_logging_data_->gloo_device_transport = parse_env("GLOO_DEVICE_TRANSPORT");
+  ddp_logging_data_->nccl_socket_ifname = parse_env("NCCL_SOCKET_IFNAME");
+  ddp_logging_data_->nccl_blocking_wait = parse_env("NCCL_BLOCKING_WAIT");
+  ddp_logging_data_->nccl_debug = parse_env("NCCL_DEBUG");
+  ddp_logging_data_->nccl_nthreads = parse_env("NCCL_NTHREADS");
+  ddp_logging_data_->nccl_ib_timeout = parse_env("NCCL_IB_TIMEOUT");
```

Comment:
Also - if we're doing environment variables, should we also eventually add backend-specific flags (such as `isHighPriorityStream` for NCCL)?

Reply:
see comments